doc/user_guide.md

```diff
@@ -435,27 +434,49 @@ Refer to spdm_server_init() in [spdm_responder.c](https://github.com/DMTF/spdm-e
 
 3. Register message process callback
 
-   This callback need handle both SPDM vendor defined message and transport layer application message.
-   ```
-   return_status libspdm_get_response_vendor_defined_request (
-     void           *spdm_context,
-     const uint32_t *session_id,
-     bool            is_app_message,
-     size_t          request_size,
-     const void     *request,
-     size_t         *response_size,
-     void           *response
-   )
-   {
-     if (is_app_message) {
-       // this is a transport layer application message
-     } else {
-       // this is a SPDM vendor defined message (without transport layer header)
-     }
-   }
-
-   libspdm_register_get_response_func (spdm_context, libspdm_get_response_vendor_defined_request);
-   ```
+   3.1 This callback handles transport layer application message.
+    ```
+    return_status libspdm_get_response (

```

Comment:
why change the function name?

Reply:
I am trying to separate normal transport layer application messages from vendor defined messages, and since 3.1 is handling only the first usescase, I renamed the callback to avoid confusion with the callback from 3.2. Now, I understand the compatibility issue for current version on LibSPDM, so I'll try to find a way for both cases to work.
The reason for this vendor defined commands dedicated methods is its importance for certain SPDM workflows.